server/eventlog.go

```diff
@@ -62,6 +83,44 @@ func contains(set [][]byte, value []byte) bool {
 	return false
 }
 
+func getVerifiedCosState(coscel cel.CEL, pcrs *tpmpb.PCRs) (*pb.AttestedCosState, error) {
+	cosState := &pb.AttestedCosState{}
+	cosState.Container = &pb.ContainerState{}
+
+	for _, record := range coscel.Records {
+		// ignore non COS CEL events
+		if !record.Content.IsCosTlv() {
+			continue
+		}
+
+		cosTlv, err := record.Content.ParseToCosTlv()
+		if err != nil {
+			return nil, err
+		}
+
+		// verify digests for the cos cel content
+		if err := cel.VerifyDigests(cosTlv, record.Digests); err != nil {
+			return nil, err
+		}
+
+		switch cosTlv.EventType {
+		case cel.ImageRefType:
+			cosState.Container.ImageReference = string(cosTlv.EventContent)
+
+		case cel.ImageDigestType:
+			cosState.Container.ImageDigest = cosTlv.EventContent
```

Comment:
It seems like you synced on Alex's changes, so this needs to be `cosState.Container.ImageDigest = string(cosTlv.EventContent)`

Reply:
yes done